test/unit_tests/braket/experimental/autoqasm/test_transpiler.py

```diff
@@ -71,50 +68,74 @@ def bell(q0: int, q1: int):
         h(q0)
         cnot(q0, q1)
 
-    @aq.main
-    def bell_decorated(q0: int, q1: int):
-        bell(q0, q1)
+    expected_partial = """OPENQASM 3.0;
+input int[32] q1;
+qubit[2] __qubits__;
+h __qubits__[1];
+cnot __qubits__[1], __qubits__[q1];"""
 
-    expected = """OPENQASM 3.0;
-qubit[4] __qubits__;
+    expected_no_arg_partial = """OPENQASM 3.0;
+qubit[2] __qubits__;
 h __qubits__[1];
 cnot __qubits__[1], __qubits__[3];"""
 
-    bell_partial = aq.main(functools.partial(bell, 1))
-    assert bell_partial(3).to_ir() == expected
+    bell_partial = aq.main(num_qubits=2)(functools.partial(bell, 1))
+    assert bell_partial.to_ir() == expected_partial
 
-    bell_decorated_partial = functools.partial(bell_decorated, 1)
-    assert bell_decorated_partial(3).to_ir() == expected
-
-    bell_noarg_partial = functools.partial(bell_decorated, 1, 3)
-    assert bell_noarg_partial().to_ir() == expected
+    bell_noarg_partial = aq.main(num_qubits=2)(functools.partial(bell, 1, 3))
+    assert bell_noarg_partial.to_ir() == expected_no_arg_partial
 
 
 def test_classmethod() -> None:
-    """Tests aq.main decorator application to a classmethod."""
+    """Tests aq.main application to a classmethod."""
+    # Note - this only works with the function call syntax, since with a decorator,
+    # the class isn't initialized fully at transpilation time.
 
     class MyClass:
         @classmethod
-        def bell(self, q0: int, q1: int):
+        def bell(cls, q0: int, q1: int):
             h(q0)
             cnot(q0, q1)
 
-        @classmethod
-        @aq.main
-        def bell_decorated(self, q0: int, q1: int):
-            self.bell(q0, q1)
+    expected = """OPENQASM 3.0;
+input int[32] q0;
+input int[32] q1;
+qubit[2] __qubits__;
+h __qubits__[q0];
+cnot __qubits__[q0], __qubits__[q1];"""
+
+    assert aq.main(num_qubits=2)(MyClass.bell).to_ir() == expected
+    assert aq.main(num_qubits=2)(MyClass().bell).to_ir() == expected
+
+
+def test_method() -> None:
+    """Tests aq.main decorator application to a classmethod."""
```

Comment:
```suggestion
    """Tests aq.main decorator application to an instance method."""
```

Reply:
Ah I actually meant to remove this test in favor of the one above it, especially since it's not even passing and @rmshaffer weighed in that we don't need to prioritize this functionality at this time